File: CSharp/HtmlTest/Base/TestWalker.cs
using HtmlWalker;
using HtmlWalker.Platform;
using System;
using System.Collections.Generic;
using System.Text;

namespace HtmlTest.Base
{
    public class TestWalker : JoinedWalker, IDisposable
    {
        public readonly TagTraceWalker tracer = new TagTraceWalker();

        public TestWalker(WalkerPlatform platform, WalkerPlatform.IOptions options = null) : base()
        {
            Output = platform.NewDocument(options);
            Walkers.Add(tracer);
        }

        public WalkerPlatform.WalkerDocument Output { get; }

        public void Dispose() => Output.Close();
    }
}

File: Java/src/htmltest/base/TestWalker.java
package htmltest.base;

import htmlwalker.CloneWalker;
import htmlwalker.FormattedCloneWalker;
import htmlwalker.JoinedWalker;
import htmlwalker.TagTraceWalker;
import htmlwalker.platform.WalkerDocument;
import htmlwalker.platform.WalkerPlatform;

public class TestWalker extends JoinedWalker
{
	private final WalkerDocument output;

    public final TagTraceWalker tracer;

    public TestWalker(WalkerPlatform platform, boolean formatted)
    {
    	super();
    	
    	output = platform.newDocument();
        tracer = new TagTraceWalker();
        
        CloneWalker cloner;
        if (formatted && !platform.isXml())
        	cloner = new FormattedCloneWalker(output);
        else
        	cloner = new CloneWalker(output);

        walkers.add(tracer);
        walkers.add(cloner);
    }
    
    public TestWalker(WalkerPlatform platform)
    {
    	this(platform, false);
    }

    public WalkerDocument output() { return output; }
}
